src/interactive-auth.js

```diff
@@ -379,7 +379,16 @@ InteractiveAuth.prototype = {
                 return flow;
             }
         }
-        return null;
+        // XXX: This happens to be the only way this can fail right now, but
+        // in general this is not really going to be an accurate error message.
+        // Ideally this would signal what inputs could be removed such that a matching
+        // flow could be found.
+        const err = new Error("No appropriate authentication flow found");
+        err.required_stages = [];
```

Comment:
I think if we're going to expect apps to catch this, we should give it a `name`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/name. `AuthenticationFlowNotFoundError` or something.

Can you also document the function(s) that will throw it?